src/formatting/expr.rs

```diff
@@ -1045,46 +1005,84 @@ impl<'a> ControlFlow<'a> {
             },
         );
 
-        let between_kwd_cond_comment = extract_comment(between_kwd_cond, context, shape);
+        let between_kwd_cond_comment = rewrite_missing_comment(between_kwd_cond, shape, context);
 
         let after_cond_comment =
-            extract_comment(mk_sp(cond_span.hi(), self.block.span.lo()), context, shape);
+            rewrite_missing_comment(mk_sp(cond_span.hi(), self.block.span.lo()), shape, context);
 
-        let block_sep = if self.cond.is_none() && between_kwd_cond_comment.is_some() {
-            ""
-        } else if context.config.control_brace_style() == ControlBraceStyle::AlwaysNextLine
+        let block_sep = if context.config.control_brace_style() == ControlBraceStyle::AlwaysNextLine
             || force_newline_brace
         {
             alt_block_sep
         } else {
             " "
         };
 
-        let used_width = if pat_expr_string.contains('\n') {
+        /* Combine with pre condition comment */
+        let label_and_keyword = &format!("{}{}", label_string, self.keyword);
+        let cond_with_between_kwd_cond_comment = if between_kwd_cond_comment
+            .as_ref()
+            .map_or(true, |c| c.is_empty())
+        {
+            format!(
+                "{}{}{}",
+                label_and_keyword,
+                if pat_expr_string.is_empty() || pat_expr_string.starts_with('\n') {
+                    ""
+                } else {
+                    " "
+                },
+                pat_expr_string,
+            )
+        } else {
+            combine_strs_with_missing_comments(
+                context,
+                &label_and_keyword,
+                &pat_expr_string,
+                between_kwd_cond,
+                shape,
+                true,
+            )
+            .unwrap()
```

Comment:
Please be sure to use the `?` operator instead of explicitly unwrapping in functions that return an Option

Reply:
Done. Changed `unwrap()` to `?` in 3 places.